test/test_tensor_ops.py

```diff
@@ -0,0 +1,411 @@
+import functools
+import torch
+
+from torch.testing._internal.common_utils import (
+    TestCase, run_tests, make_tensor, suppress_warnings)
+
+from torch.testing._internal.common_device_type import (
+    instantiate_device_type_tests, onlyOnCPUAndCUDA, ops)
+
+from torch.testing._internal.common_methods_invocations import (
+    OpInfo, SampleInput)
+
+
+class TensorCreationOpInfo(OpInfo):
+    """OpInfo Base class for tensor creation operations.
+    """
+
+    all_dtypes = torch.testing._dispatch_dtypes(torch.testing.get_all_dtypes())
+    all_dtypes_numpycompat = torch.testing._dispatch_dtypes(torch.testing.get_all_dtypes(
+        include_half=False,
+        include_bfloat16=False,
+        include_complex32=False))
+    all_dtypes_grad = torch.testing._dispatch_dtypes(torch.testing.floating_and_complex_types())
+
+    def __init__(self,
+                 name,
+                 *,
+                 dtypes=None,
+                 supports_autograd=None,
+                 **kwargs):
+        supports_autograd = False if supports_autograd is None else supports_autograd
+        if supports_autograd:
+            dtypes = self.all_dtypes_grad if dtypes is None else dtypes
+        else:
+            dtypes = self.all_dtypes if dtypes is None else dtypes
+        super().__init__(name,
+                         dtypes=dtypes,
+                         supports_autograd=supports_autograd,
+                         check_batched_grad=False,
+                         check_batched_gradgrad=False,
+                         **kwargs)
+
+    def get_devices(self, skip_device=None, max_device_indices=2):
```

Comment:
We already have a lot of support for enumerating devices. Why does this need another function?